cloud/libvirt/actuators/machine/utils/volume.go

```diff
@@ -196,46 +194,55 @@ func CreateVolume(volumeName, poolName, baseVolumeID, source, volumeFormat strin
 	return nil
 }
 
-// DeleteVolume removes the volume identified by `key` from libvirt
-func DeleteVolume(name string, client *Client) error {
-	volumePath := fmt.Sprintf(baseVolumePath+"%s", name)
-	volume, err := client.libvirt.LookupStorageVolByPath(volumePath)
+func getVolumeFromPool(volumeName, poolName string, virConn *libvirt.Connect) (*libvirt.StorageVol, error) {
+	pool, err := virConn.LookupStoragePoolByName(poolName)
 	if err != nil {
-		return fmt.Errorf("Can't retrieve volume %s", volumePath)
+		return nil, fmt.Errorf("can't find storage pool %q: %v", poolName, err)
 	}
-	defer volume.Free()
+	defer pool.Free()
 
 	// Refresh the pool of the volume so that libvirt knows it is
 	// not longer in use.
-	volPool, err := volume.LookupPoolByVolume()
+	waitForSuccess("error refreshing pool for volume", func() error {
+		return pool.Refresh(0)
```

Comment:
Can this run forever?

Reply:
I'm new to Go/Installer so maybe I'm missing something but `pool.Refresh` call is blocking and since it does I/O, theoretically it can take a long time before returning. If the call returns an error, it's very unlikely to succeed if called again. So I'm not sure looking over it until it succeeds is the way to go here.